File: API/Models/Vehicle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Models
{
    public class Vehicle
    {
        public long VehicleId {get; set;}

        public string Type {get;set;}

        public string Brand {get;set;}

        public string Model {get;set;}

        public string Fuel{get; set;}

        public string Gearbox {get; set;}


        public string Color {get;set;}

        public int Power {get;set;}

        public float Engine{get;set;}

        public bool IsCrashed{get;set;}

        public int Mileage{get;set;}

        public int ProductionYear{get;set;}

        public ICollection<Auction> Auctions {get; set;}

        public ICollection<Advertisment> Advertisments {get; set;}

        public ICollection<Photo> Photos {get; set;}

        public long UserId {get; set;}

        public User User {get; set;}
        }
}

File: client/src/app/_models/vehicle.ts
export interface Vehicle {
  type: string;
  brand: string;
  model: string;
  fuel: string;
  gearbox: string;
  color: string;
  power: number;
  engine: number;
  isCrashed: boolean;
  mileage: number;
  productionYear: number;
  userId: number;
}
export class VehicleClass implements Vehicle {
  type: string = '';
  brand: string = '';
  model: string = '';
  color: string = '';
  fuel: string = '';
  gearbox: string = '';
  power: number = null;
  engine: number = null;
  isCrashed: boolean = false;
  mileage: number = null;
  productionYear: number = null;
  userId: number;
  constructor() {}
}
export class TypeClass {
  typeList: any = [
    'SUV',
    'Combi',
    'Sedan',
    'Hatchback',
    'Coupe',
    'Cabriolet',
    'Motorcycle',
    'Truck',
    'Mini Bus',
  ];
  constructor() {}
}

export class BrandClass {
  brandList: any = [
    'BMW',
    'Audi',
    'Volkswagen',
    'Ford',
    'Opel',
    'Mercedes-Benz',
    'Skoda',
    'Renault',
    'Citroen',
    'Peugeot',
    'Alfa Romeo',
    'Mazda',
    'Dodge',
    'Chevrolet',
    'Honda',
    'Lexus',
    'Nissan',
    'Pontiac',
    'Lincoln',
    'Cadillac',
    'Fiat',
    'Ferrari',
    'Lamborghini',
    'Aston Martin',
    'Bentley',
    'Porsche',
    'Rolls-Royce',
    'Land-Rover',
    'Kia',
    'Chrysler',
    'Seat',
    'Tesla',
    'Subaru',
    'Hyundai',
    'Dacia',
    'Mitsubishi',
    'Lancia',
  ];
}
export class ColorClass {
  colorList: any = [
    'black',
    'white',
    'red',
    'yellow',
    'green',
    'silver',
    'grey',
    'blue',
    'orange',
    'purple',
    'pink',
  ];
}

export class FuelClass {
  fuelList: any = ['Petrol', 'Diesel', 'Gas'];
}
